### DISCUSSION TITLE
x height for accessibility

### DISCUSSION THREAD

**Discussion by Myndex** [seek-oss/capsize#40] [Ideas]
This looks promising.

I'd like to mention that x-height is the **single most critical font metric for readability**, especially for blocks of body text.

Associated with contrast, we are working on standards/guidelines using x-height as the defining or reference size metric.

I thought it might be useful to mention this early on.

Thank you!

Andy

**Comment by apkoponen** [comment]
capsize does already extract the xHeight metric from the fonts and I imagine it would be possible to add an accessibility warning to the  site when the xHeight is too small. Is there a reference, where valid xHeight values could be found?

**Comment by Myndex** [comment]
> capsize does already extract the xHeight metric from the fonts and I imagine it would be possible to add an accessibility warning to the site when the xHeight is too small. Is there a reference, where valid xHeight values could be found?

Hi @apkoponen 

It is a work in progress, but by the time Silver is a first public working draft, we "hope" to have that in place. At present we are using "50% of body height" for the interim breakpoints. Eventually there is a goal of a workable CSS property, but it's "not ready for prime time"  LOL.

**Comment by michaeltaranto** [comment]
Hey @Myndex 👋 

There's definitely room for this kind of validation coming in at some stage. Based on the specified capHeight/fontSize, if the resulting x-height is of a certain size we could tell you. Are there any associated guideline drafts you could share?

**Reply by Myndex** [reply]
> Hey @Myndex 👋
> 
> There's definitely room for this kind of validation coming in at some stage. Based on the specified capHeight/fontSize, if the resulting x-height is of a certain size we could tell you. Are there any associated guideline drafts you could share?

Hi @michaeltaranto 

The Visual Contrast specification for WCAG 3.0 (draft, aka "Silver") indicates the importance of x-height, and some of that is indicated on the contrast tool for APCA. See https://www.myndex.com/APCA/ which also has the draft specifications.

**Comment by andriytyurnikov** [comment]
@Myndex here is a Critical Print Size calculator that takes into account font metrics, distance to the eye, and visual acuity of the audience https://andriytyurnikov.me/garage/font-calc
